codejail/languages.py

```diff
@@ -0,0 +1,138 @@
+"""
+Language-specific configuration for codejails.
+
+A language consists of:
+
+    `name` A human readable name for the language.
+
+    `argv` A list of extra language-arguments to pass to the Jail's
+    executable.
+
+    `safe_exec_template` Wrapper code to make safe_exec work properly.  If a
+    language has safe_exec_template set to None, commands configured to use
+    that language will not be able to use `codejail.jail.Jail.safe_exec`, and
+    will need to use `codejail.jail.Jail.jail_code` instead.
+
+Preconfigured `Language` objects exist for `python2`, `python3`, and `other`, but
+applications can define their own languages by instantiating `Language`.
+"""
+
+from collections import namedtuple
+import textwrap
+
+
+Language = namedtuple('Language', ['name', 'argv', 'safe_exec_template'])
+
+
+# Allow lowercased names at module level
+# pylint: disable=invalid-name
+
+other = Language(
+    name='other',
+    argv=[],
+    safe_exec_template=None,
+)
+
+python2 = Language(
+    name='python2',
+    argv=[
+        '-E',  # Ignore the environment variables PYTHON*
+        '-B',  # Don't write .pyc files.
+    ],
+    safe_exec_template=textwrap.dedent(
```

Comment:
What would the alternative be?  It can't just be a called python function, because it's not executed in-process.  The python3 safe_exec_template would fail to import with syntax errors if it were inlined, and once (if) we expand safe_exec to work with other languages, the template would not even resemble python code.